source/common/mongo/proxy.cc

```diff
@@ -182,6 +184,28 @@ void ProxyFilter::decodeReply(ReplyMessagePtr&& message) {
     active_query_list_.erase(i);
     break;
   }
+
+  if (active_query_list_.empty() && drain_decision_.drainClose()) {
+    ENVOY_LOG(debug, "drain closing mongo connection");
+    stats_.cx_drain_close_.inc();
+
+    // We are currently in the write path, so we need to let the write flush out before we close.
+    // We do this by creating a timer and firing it with a zero timeout. This will cause it to run
+    // in the next event loop iteration. This is really a hack. A better solution would be to
+    // introduce the concept of a write complete callback so we can get notified when the write goes
+    // out (e.g., flow control, further filters, etc.). This is a much larger project so we can
+    // start with this since it will get the job done.
+    // TODO(mattklein123): Investigate a better solution for write complete callbacks.
+    if (drain_close_timer_ == nullptr) {
+      drain_close_timer_ =
+          read_callbacks_->connection().dispatcher().createTimer([this] { onDrainClose(); });
```

Comment:
This is basically a `post()`, right? Would it work and/or be any clearer to do that instead?

Reply:
Posts cannot be canceled currently. We need to be able to cancel this.